generator/internal/discovery_type_vertex.h

```diff
@@ -34,9 +34,11 @@ namespace generator_internal {
 // of edges are unidirectional, and both of the resulting graphs are acyclic.
 class DiscoveryTypeVertex {
  public:
-  DiscoveryTypeVertex();
+  // descriptor_pool should never be NULL, and the constructor will assert if it
+  // is.
```

Comment:
It doesn't look like the constructor does this.  Indeed, it ignores the `DescriptorPool` argument.